src/main/java/com/github/mizool/jediscache/JedisCache.java

```diff
@@ -0,0 +1,353 @@
+        return result;
+    }
+
+    @Override
+    public void removeAll(Set<? extends K> keys)
+    {
+        keys.forEach(this::remove);
+    }
+
+    @Override
+    public void removeAll()
+    {
+        verifyOpen();
+        Set<byte[]> keys;
+        try (Jedis jedis = jedisPool.getResource())
+        {
+            keys = jedis.hkeys(jedisCacheName);
+        }
+        keys.stream().map(keyConverter::toKey).forEach(this::remove);
+    }
+
+    @Override
+    public void clear()
+    {
+        verifyOpen();
+        try (Jedis jedis = jedisPool.getResource())
+        {
+            jedis.del(jedisCacheName);
+        }
+    }
+
+    @Override
+    public <C extends Configuration<K, V>> C getConfiguration(Class<C> configurationClass)
+    {
+        return configurationClass.cast(configuration);
+    }
+
+    @Override
+    public <T> T invoke(K key, EntryProcessor<K, V, T> entryProcessor, Object... arguments)
+        throws EntryProcessorException
+    {
+        throw new UnsupportedOperationException("processor support is not implemented");
+    }
+
+    @Override
+    public <T> Map<K, EntryProcessorResult<T>> invokeAll(
+        Set<? extends K> keys, EntryProcessor<K, V, T> entryProcessor, Object... arguments)
+    {
+        throw new UnsupportedOperationException("processor support is not implemented");
+    }
+
+    @Override
+    public void close()
+    {
+        if (!closed.getAndSet(true))
+        {
+            cacheManager.releaseCache(name);
+        }
+    }
+
+    @Override
+    public boolean isClosed()
+    {
+        return closed.get();
+    }
+
+    @Override
+    public <T> T unwrap(Class<T> classOfT)
+    {
+        if (classOfT.isAssignableFrom(getClass()))
+        {
+            return classOfT.cast(this);
+        }
+
+        throw new IllegalArgumentException("Unwapping to " + classOfT + " is not a supported by this implementation");
+    }
+
+    @Override
+    public void registerCacheEntryListener(CacheEntryListenerConfiguration<K, V> cacheEntryListenerConfiguration)
+    {
+        throw new UnsupportedOperationException("listener support is not implemented");
+    }
+
+    @Override
+    public void deregisterCacheEntryListener(CacheEntryListenerConfiguration<K, V> cacheEntryListenerConfiguration)
+    {
+        throw new UnsupportedOperationException("listener support is not implemented");
+    }
+
+    @Override
+    public Iterator<Entry<K, V>> iterator()
+    {
+        verifyOpen();
+        ScanParams scanParams = new ScanParams();
+        byte[] cursor = ScanParams.SCAN_POINTER_START_BINARY;
+        ScanResult<Map.Entry<byte[], byte[]>> scanResult;
+        try (Jedis jedis = jedisPool.getResource())
+        {
+            scanResult = jedis.hscan(jedisCacheName, cursor, scanParams);
+        }
+        return scanResult.getResult().stream().map(this::turnIntoEntries).iterator();
+    }
+
+    private Entry<K, V> turnIntoEntries(Map.Entry<byte[], byte[]> entry)
```

Comment:
the method takes a `Map.Entry` and returns a `Cache.Entry`, so the name should be slightly more specific than this (also, it uses plural but is singular). how about `deserialize`?